src/Billing/AutoBilling.php

```diff
@@ -0,0 +1,336 @@
+         * like doing billing from the calendar
+         * @returns int
+         */
+       //$enc = $this->getEventEncounter($pid, $event_date);
+        $enc = $_SESSION['encounter'];
+        /**
+         * Grab the provider for the encounter from the event table
+         */
+        //$provider = $this->getEventProvider($pid, $event_date);
+        $provider = $_SESSION['authUserID'];
+        /**
+         * find out if there are any entries in the billing table for event date for this patient
+         * moved to calendar inc
+         *
+         */
+        $hasBilling = $this->findEventBilling($this->pid, $enc);
+
+        /**
+         * Gets the modifier assigned to the
+         * If there are modifiers in the code table
+         */
+        $modifier = $this->getModifier($code);
+        $modifier = str_replace(",", "", $modifier);
+
+        //insert billing if there is note and entry for this particular encounter
+        if (empty($hasBilling['encounter'])) {
+            if (!empty($code)) {
+                try {
+                    $codetype = "CPT4";
+                    $this->insertCPTBilling($event_date, $code, $this->pid, $provider, $this->userid, $enc, $text, $fees, $icd10, $codetype, $modifier);
+                } catch (Exception $e) {
+                    return $e;
+                }
+            }
+
+            try {
+                //Enter ICD10 if it exist
+                if (!empty($desc) && empty($res)) {
+                    $codetype = "ICD10";
+                    $this->insertICDBilling($event_date, $codetype, $icd10, $this->pid, $provider, $this->userid, $enc, $desc);
+                }
+            } catch (Exception $e) {
+                return $e;
+            }
+        }
+    }// End of billing entries
+
+    /**
+     * @param $event_date
+     * @param $code
+     * @param $pid
+     * @param $provider
+     * @param $userid
+     * @param $enc
+     * @param $text
+     * @param $fees
+     * @param $icd
+     * @param $codetype
+     * @param $modifier
+     */
+    private function insertCPTBilling($event_date, $code, $pid, $provider, $userid, $enc, $text, $fees, $icd10, $codetype, $modifier)
+    {
+        $sql = "REPLACE INTO billing SET " .
+            "date = ? , " .
+            "code_type = ?, " .
+            "code = ?, " .
+            "pid = ?, " .
+            "provider_id = ?, " .
+            "user = ?, " .
+            "groupname = 'default', " .
+            "authorized = '1', " .
+            "encounter = ?, " .
+            "code_text = ?, " .
+            "billed = '0', " .
+            "activity = '1', " .
+            "modifier = ?, " .
+            "units = '1', " .
+            "fee = ?, " .
+            "justify = ?, " .
+            "pricelevel = 'standard'";
+
+        try {
+             $justify = "ICD10|$icd10:";
+              sqlStatement($sql, [$event_date, $codetype, $code, $pid, $provider, $userid, $enc, $text, $modifier, $fees, $justify]);
```

Comment:
Will this even throw an error if it breaks?
(btw, @adunsulag has created a class for sql call that will throw errors)

Reply:
Great, where is it or are you going to make me hunt for it?

Reply:
hi @juggernautsei, `sql.inc`, `function sqlStatementThrowException($statement, $binds = false)`

Reply:
You might want to look here also https://github.com/openemr/openemr/blob/master/src/Common/Database/QueryUtils.php